### PR TITLE
WRO-3956:Upgrading from Gatsby 2 to 4

### PR DESCRIPTION
**Pull request by sjro**
### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Need to upgrade the enact libraries and the gatsby modules. (from gatsby 2 to gatsby 4)
- Cannot query field "context" on type "SitePage".  error 
- ModuleParseError during building development bundle

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
- Update dependencies
- Update a .travis.yml file since gatsby 4 works node 14 or higher version.
- Fix it by the link (https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v3-to-v4/#changes-to-built-in-types) since SitePage.context is no longer abailable in GraphQL queries.
- Update a gatsby-config.js file to change the default css-loader option of gatsby-plugin-less and gatsby-plugin-postcss
- Remove pathContext as described int this link ( https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v2-to-v3/)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,15 @@
+dist: focal
 language: node_js
 node_js:
-    - 12
+    - lts/*
+    - 18
```
Comment (0x64) [inline .travis.yml:5]: Do we list two versions under `node_js` option?

File: gatsby-node.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 /* eslint-env node */
 // const GracefulFSPlugin = require('graceful-fs-webpack-plugin');
 // const autoprefixer = require('autoprefixer');
+//const FilterWarningsPlugin = require("webpack-filter-warnings-plugin");
```
Comment (0x64) [inline gatsby-node.js:4]: Is this required to be here?

File: gatsby-node.js [hunk 3]
```diff
@@ -30,7 +32,11 @@ exports.onCreateWebpackConfig = ({
 					}
 				}),
 				ignore: () => new webpack.IgnorePlugin(/^(xor|props)$/)
-			})
+			}),
+			/*new FilterWarningsPlugin({
```
Comment (0x64) [inline gatsby-node.js:36]: Hmm... If there is no reason, let's remove this commented-out new code block.

File: sample-runner/shared/EnactLiveEdit.module.less [hunk 4]
```diff
@@ -1,4 +1,4 @@
-// EnactLiveEdit.less
+// EnactLiveEdit.module.less
```
Comment (0x64) [inline sample-runner/shared/EnactLiveEdit.module.less:1]: This change means that `.error` and `.sandbox` are not for global scope. Do I understand correctly?

File: src/components/DocsNav/DocsNav.js [hunk 5]
```diff
@@ -46,7 +46,7 @@ const DocsNav = kind({
 			if (page) {
 				const {title, description} = pageMetadata(page.node.path, jsMetadata);
 				return {
-					title: page.node.context.title || title,
+					title: title,
```
Comment (0x64) [inline src/components/DocsNav/DocsNav.js:49]: Could you please explain this change?

File: .travis.yml (hunk 1 above)
Comment (sjro) [reply .travis.yml]: yes. we can test on multiple node versions

**Pull request merged**